### PR TITLE
Support gene symbols in study-view api's

### PR DESCRIPTION
**Pull request by kalletlak**
# What? Why?
Part of https://github.com/cBioPortal/cbioportal/issues/4983#issuecomment-454933667

### CODE REVIEW

File: src/pages/studyView/virtualStudy/VirtualStudy.tsx [hunk 1]
```diff
@@ -174,8 +174,9 @@ export default class VirtualStudy extends React.Component<IVirtualStudyProps, {}
 
     readonly genes = remoteData({
         invoke: async () => {
-            if(!_.isEmpty(this.entrezIds)){
-                return client.fetchGenesUsingPOST({geneIdType: "ENTREZ_GENE_ID", geneIds: this.entrezIds as any});
+            if(!_.isEmpty(this.geneIds)){
+                // not all geneIds are ENTREZ_GENE_ID
```
Comment (zhx828) [inline src/pages/studyView/virtualStudy/VirtualStudy.tsx]: Do we need to make separate calls for gene ids that are entrez_gene_id?

File: src/pages/studyView/UserSelections.tsx [hunk 2]
```diff
@@ -193,11 +192,11 @@ export default class UserSelections extends React.Component<IUserSelectionsProps
                 components={this.props.filter.cnaGenes.map(filter => {
                     return <GroupLogic
                         components={filter.alterations.map(filter => {
-                            const hugoGeneSymbol = this.props.getSelectedGene(filter.entrezGeneId);
+                            const hugoGeneSymbol = this.props.getSelectedGene(filter.geneId);
```
Comment (zhx828) [inline src/pages/studyView/UserSelections.tsx]: Need to check getSelectedGene

File: src/pages/studyView/table/CNAGenesTable.tsx [hunk 3]
```diff
@@ -229,10 +229,10 @@ export class CNAGenesTable extends React.Component<ICNAGenesTablePros, {}> {
                 },
                 render: (data: CopyNumberCountByGeneWithCancerGene) => (
                     <LabeledCheckbox
-                        checked={this.isChecked(data.entrezGeneId, data.alteration)}
-                        disabled={this.isDisabled(data.entrezGeneId, data.alteration)}
+                        checked={this.isChecked(data.hugoGeneSymbol, data.alteration)}
```
Comment (zhx828) [inline src/pages/studyView/table/CNAGenesTable.tsx]: I think we should still use entrezGeneId for filtering

File: src/pages/studyView/virtualStudy/VirtualStudy.tsx [hunk 4]
```diff
@@ -174,8 +174,9 @@ export default class VirtualStudy extends React.Component<IVirtualStudyProps, {}
...
+                return client.fetchGenesUsingPOST({geneIdType: "HUGO_GENE_SYMBOL", geneIds: this.geneIds as any});
```
Comment (zhx828) [inline src/pages/studyView/virtualStudy/VirtualStudy.tsx]: I'm actually confused, I thought the gene id represents entrezGeneId or HugoSymbol. But this api call does not reflect on that.

**Comment by zhx828** [comment]
@kalletlak another update to my previous comment. I see the effort try to use geneId in the filtering since we support that the backend. The most places we use geneId should be to the backend endpoint to let it resolve hugosymbol/entrezGeneId. In the actual table component, since the data it consumes should always have entrezGeneId, then let's use that as the uniqueId instead of geneId.

File: src/pages/studyView/virtualStudy/VirtualStudy.tsx (hunk 4 above)
Comment (kalletlak) [reply src/pages/studyView/virtualStudy/VirtualStudy.tsx]: its a complete different API. not part of study view API's

File: src/pages/studyView/UserSelections.tsx [hunk 5]
```diff
@@ -35,7 +33,7 @@ export interface IUserSelectionsProps {
     updateCustomChartFilter: (chartMeta: ChartMeta, values: string[]) => void;
     clearGeneFilter: () => void;
     clearCNAGeneFilter: () => void;
-    removeGeneFilter: (entrezGeneId: number) => void;
+    removeGeneFilter: (geneId: string) => void;
```
Comment (zhx828) [inline src/pages/studyView/UserSelections.tsx]: I guess here should be `hugoGeneSymbol`

File: src/pages/studyView/table/CNAGenesTable.tsx [hunk 6]
```diff
@@ -401,7 +393,7 @@ export class CNAGenesTable extends React.Component<ICNAGenesTablePros, {}> {
     isSelectedRow(data: CopyNumberCountByGeneWithCancerGene) {
         return !_.isUndefined(
             _.find(_.union(this.selectedRows, this.preSelectedRows), function(row) {
-                return row.entrezGeneId === data.entrezGeneId && row.alteration === data.alteration;
+                return row.hugoGeneSymbol === `${data.hugoGeneSymbol}` && row.alteration === data.alteration;
```
Comment (zhx828) [inline src/pages/studyView/table/CNAGenesTable.tsx]: why are we wrapping it?

File: src/pages/studyView/virtualStudy/VirtualStudy.tsx [hunk 7]
```diff
@@ -245,7 +205,7 @@ export default class VirtualStudy extends React.Component<IVirtualStudyProps, {}
                                             onChange={event => this.description = event.currentTarget.value}
                                         />
                                     </div>
-                                    <span style={{ 'display': 'block', 'font-weight': 'bold' }}>This virtual study was derived from:</span>
+                                    <span style={{ 'display': 'block', 'fontWeight': 'bold' }}>This virtual study was derived from:</span>
```
Comment (zhx828) [inline src/pages/studyView/virtualStudy/VirtualStudy.tsx]: You need to wrap these keys. But it's really minor.

**Pull request merged**